mesonbuild/minstall.py

```diff
@@ -244,6 +244,27 @@ def restore_selinux_contexts() -> None:
                   'Standard output:', out.decode(),
                   'Standard error:', err.decode(), sep='\n')
 
+def apply_ldconfig() -> None:
+    '''
+    Apply ldconfig to update the ld.so.cache.
+
+    If $DESTDIR is set, do not execute ldconfig.
+    '''
+    if not shutil.which('ldconfig'):
+        # If we don't have ldconfig, failure is ignored quietly.
+        return
+
+    if os.environ.get('DESTDIR'):
+        print('Do not execute ldconfig if DESTDIR is set')
+        return
+
+    with subprocess.Popen(['ldconfig'],
+                          stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as proc:
+        out, err = proc.communicate(input=b'\0'.join(os.fsencode(f) for f in selinux_updates) + b'\0')
```

Comment:
You should use `Popen_safe()` which does all kind of hacks to get output encoding right. Also you use `selinux_updates` here, seems copy/paste error.

Reply:
While at it, I'm wondering if we should fix the selinux part too:
- Skip it completely when DESTDIR is set, instead of just ignoring error.
- Use `proc, out, err = Popen_safe(b'\0'.join(os.fsencode(f) for f in selinux_updates) + b'\0')`

Reply:
Same PR but different commit !